src/NodeHelper.groovy

```diff
@@ -0,0 +1,1136 @@
+
+        Computer computer = getComputer(computerName);
+        if (computer != null) {
+            def kernelInfo = getOsKernelInfo(computer.getName());
+
+            switch(kernelInfo.get(0)) {
+                case "linux":
+                    ret = getPlatformString(getOsArch(computer.getName())) + kernelInfo.get(0);
+                    break;
+                case "windows":
+                    ret = "win";
+                    break;
+                case "mac":
+                case "aix":
+                case "zos":
+                    ret = kernelInfo.get(0);
+                    break;
+            }
+
+            ret = "hw.platform." + ret;
+        }
+
+        return ret.toLowerCase();
+    }
+
+    private String getPlatformString(String value) {
+        String ret = "getPlatformString:INVALID_PLATFORM ${value}";
+
+        if (value.length() > 1) {
+            switch (value) {
+                case "amd64":
+                    ret = "x";
+                    break;
+                case "ppc64":
+                case "ppc64le":
+                    ret = "p";
+                    break;
+                case "s390x":
+                    ret = "z";
+                case "arm":
+                    ret = "arm";
+                    break;
+            }
+        }
+
+        return ret;
+    }
+
+    public String constructArchLabel(String computerName) {
+        String ret = "constructArchLabel:COMPUTER_NOT_FOUND";
+
+        Computer computer = getComputer(computerName);
+        if (computer != null) {
+            String arch = getOsArch(computer.getName());
+
+            switch (arch) {
+                case "x86_64":
+                case "amd64":
+                    ret = "x86";
+                    break;
+                case "ppc64":
+                case "ppc64le":
+                    ret = "ppc";
+                    break;
+                case "s390x":
+                    ret = "s390";
+                    break;
+                case "arm":
+                    ret = arch;
+                    break;
+                default:
+                    ret = "INVALID_ARCH";
+                    break;
+            }
+            ret = "hw.arch." + ret;
+        }
+
+
+        return ret.toLowerCase();
+    }
+
+    public String constructKernelLabel(String computerName) {
+        String ret = "constructKernelLabel:COMPUTER_NOT_FOUND";
+
+        Computer computer = getComputer(computerName);
+        if (computer != null) {
+            def kernelInfo = getOsKernelInfo(computer.getName());
+            ret = "sw.os." + kernelInfo.get(0);
+        }
+
+        return ret;
+    }
+
+    public String constructHypervisorLabel(String computerName) {
+        String ret = "constructHypervisorLabel:COMPUTER_NOT_FOUND";
+
+        Computer computer = getComputer(computerName);
+        if (computer != null) {
+            ret = "hw.hypervisor.";// TODO: finish implementation, get something
```

Comment:
Add issue in issue tracker

Reply:
Done! [Implement logic to get Hypervisor information](https://github.com/AdoptOpenJDK/openjdk-jenkins-helper/issues/4)